src/test/scala/com/databricks/labs/mosaic/expressions/geometry/ST_SimplifyTest.scala

```diff
@@ -0,0 +1,34 @@
+package com.databricks.labs.mosaic.expressions.geometry
+
+import com.databricks.labs.mosaic.core.geometry.api.GeometryAPI.{ESRI, JTS}
+import com.databricks.labs.mosaic.core.index.{BNGIndexSystem, H3IndexSystem}
+import org.apache.spark.sql.QueryTest
+import org.apache.spark.sql.catalyst.expressions.CodegenObjectFactoryMode
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.test.SharedSparkSession
+
+class ST_SimplifyTest extends QueryTest with SharedSparkSession with ST_SimplifyBehaviors {
+
+    private val noCodegen =
+        withSQLConf(
+          SQLConf.WHOLESTAGE_CODEGEN_ENABLED.key -> "false",
+          SQLConf.CODEGEN_FACTORY_MODE.key -> CodegenObjectFactoryMode.NO_CODEGEN.toString
+        ) _
+
+    private val codegenOnly =
+        withSQLConf(
+          SQLConf.ADAPTIVE_EXECUTION_ENABLED.key -> "false",
+          SQLConf.WHOLESTAGE_CODEGEN_ENABLED.key -> "true",
+          SQLConf.CODEGEN_FACTORY_MODE.key -> CodegenObjectFactoryMode.CODEGEN_ONLY.toString
+        ) _
+
+    test("Testing stSimplify (H3, JTS) NO_CODEGEN") { noCodegen { simplifyBehavior(H3IndexSystem, JTS) } }
+    test("Testing stSimplify (H3, ESRI) NO_CODEGEN") { noCodegen { simplifyBehavior(H3IndexSystem, ESRI) } }
```

Comment:
Here also we don't need to test across grid systems

Reply:
This only tests against the (real) H3 grid index system. I'm working on a separate PR to use a mock index system for these kind of tests.